ludwig/schema/trainer.py

```diff
@@ -153,7 +153,7 @@ class ECDTrainerConfig(BaseTrainerConfig):
         ),
         parameter_metadata=TRAINER_METADATA["batch_size"],
         field_options=[
-            schema_utils.PositiveInteger(default=DEFAULT_BATCH_SIZE, description=""),
+            schema_utils.PositiveInteger(default=DEFAULT_BATCH_SIZE, description="", allow_none=False),
```

Comment:
Example of what I meant in the thread last week - I think it's a better user experience to be prescriptive about `null` rather than non-`null`-ity.

Reply:
I can go either way on what should be the default for `allow_none` -- there seems to be a fair number of parameters on either side.

On one hand, we make it so that parameter definers need to be explicit about when None is a legitimate value while on the other, we make it so that parameter definers need to be explicit about when None is not a legitimate value. Theoretically, these should be the same requirement!

Whichever way we go, parameters with non-None defaults should be set with `allow_none=False`, unless `None` is a non-trivial legitimate value, in which case they should bet set with `allow_none=True`. Happy to discuss more / follow up in a separate PR.